launcher/firstLaunch/firstlaunch_moc.cpp

```diff
@@ -279,7 +293,69 @@ QString FirstLaunchView::getHeroesInstallDir()
 	return QString{};
 }
 
-void FirstLaunchView::copyHeroesData(const QString & path)
+void FirstLaunchView::extractGogData()
+{
+#ifndef VCMI_NO_INNOEXTRACT
+	QString filterExe = tr("GOG executable") + " (*.exe)";
+	QString fileExe = QFileDialog::getOpenFileName(this, tr("Select a GOG installer (exe) file..."), QDir::homePath(), filterExe);
+	if(fileExe.isEmpty())
+		return;
+
+	QString filterBin = tr("GOG bin file") + " (*.bin)";
+	QString fileBin = QFileDialog::getOpenFileName(this, tr("Select a GOG data (bin) file..."), QFileInfo(fileExe).absolutePath(), filterBin);
+	if(fileBin.isEmpty())
+		return;
+
+	if(QFileInfo(fileExe).completeBaseName() + QString("-1") != QFileInfo(fileBin).completeBaseName())
+	{
+		QMessageBox::critical(this, tr("File name error!"), tr("The filename of exe and bin has to be compatible!"), QMessageBox::Ok, QMessageBox::Ok);
+		return;
+	}
+
+	QTimer::singleShot(100, this, [this, fileExe, fileBin](){ // background to make sure FileDialog is closed...
+		QTemporaryDir dir;
+		if(dir.isValid() && QFile::exists(fileExe) && QFile::exists(fileBin)) {
+			ui->pushButtonGogInstall->setText(tr("Installing... Please wait!"));
+			QPalette pal = ui->pushButtonGogInstall->palette();
+			pal.setColor(QPalette::Button, QColor(Qt::yellow));
+			ui->pushButtonGogInstall->setAutoFillBackground(true);
+			ui->pushButtonGogInstall->setPalette(pal);
+			ui->pushButtonGogInstall->update();
+			ui->pushButtonGogInstall->repaint();
+
+			QString tmpFileExe = dir.filePath(QFileInfo(fileExe).fileName());
+			QFile(fileExe).copy(tmpFileExe);
+			QFile(fileBin).copy(dir.filePath(QFileInfo(fileBin).fileName()));
+
+			::extract_options o;
+			o.extract = true;
+
+			// standard settings
+			o.gog_galaxy = true;
+			o.codepage = (uint32_t)0;
+			o.output_dir = dir.path().toStdString();
+			o.extract_temp = true;
+			o.extract_unknown = true;
+			o.filenames.set_expand(true);
+
+			o.preserve_file_times = true; // also correctly closes file -> without it: on Windows the files are not written completly
+
+			process_file(tmpFileExe.toStdString(), o);
+
+			QStringList dirData = QDir(dir.path()).entryList({"data"}, QDir::Filter::Dirs);
```

Comment:
I think our current file copy algorithm actually looks up directories to copy in case-insensitive way. Perhaps use same approach here?

Reply:
I've copied the approach from here: https://github.com/vcmi/vcmi/blob/de9dbf0cd78c300dcd1663fc41615b52aadcf0b4/launcher/firstLaunch/firstlaunch_moc.cpp#L380

Basically only to check if we have some H3-like output from innoextract (as you can also feed it with other games for example).